wordpress-mobile/WordPress-iOS
# Gutenberg: Adjust font sizes in `RichText` based on device settings

Partial fix for: https://github.com/wordpress-mobile/gutenberg-mobile/issues/4097

> **Note**: There are times where the font appears overly large when used with custom fonts and block-based themes, this will be addressed in a separate Gutenberg PR for _both_ platforms. The scope of this specific PR, in conjunction with the Aztec PR, is to enable support for Dynamic Type in all `RichText` components. 

## Related PRs

* Aztec: https://github.com/wordpress-mobile/AztecEditor-iOS/pull/1381
* Gutenberg Mobile: https://github.com/wordpress-mobile/gutenberg-mobile/pull/6465

## To Test

- Navigate to the **Accessibility** section in the **Settings** app. 
- Select **Display & Text Size** then **Larger Accessibility Sizes** to on.
- Select the largest possible accessibility size.
- Next, open the Jetpack or WordPress app and select a site with a block-based theme.
- Navigate to the editor in the app.
- Either add a heading block or begin typing into a paragraph block.
- Select the block's cog/gear icon to open its settings and select the largest available font size.
- Experiment with typing more text and adding more heading or paragraph blocks.
- Verify that the font has been scaled in accordance with your device and block settings, and that it is not unreasonably large.

## Screenshots 

<details>
 <summary>Before vs. after ⤵️</summary>

| Before  | After |
| ------------- | ------------- |
| <img src="https://github.com/wordpress-mobile/WordPress-iOS/assets/2998162/f3d8ad72-fe4f-429b-aafc-a08c5e7a3d91" width="100%"> | <img src="https://github.com/wordpress-mobile/WordPress-iOS/assets/2998162/e54e12a6-e11b-4cc2-868e-ff72e5a6a8d2" width="100%"> |
</details>

<details>
 <summary>Android comparison ⤵️</summary>

| Android  | iOS |
| ------------- | ------------- |
| <img src="https://github.com/wordpress-mobile/WordPress-iOS/assets/2998162/14f283ec-0700-47fd-9ab2-3f7b07a1e8ef" width="100%"> | <img src="https://github.com/wordpress-mobile/WordPress-iOS/assets/2998162/e54e12a6-e11b-4cc2-868e-ff72e5a6a8d2" width="100%"> |
</details>

<hr />

## Regression Notes
1. Potential unintended areas of impact

It would be unintended for this PR to change the way font sizes are handled outside of Dynamic Type adjustments. 

2. What I did to test those areas of impact (or what existing automated tests I relied on)

- Manually tested changes.
- Relied on existing automated tests for [Aztec](https://github.com/wordpress-mobile/AztecEditor-iOS/tree/006aceaaf1d2718113263a1e3346180ae1133403/AztecTests) and [the RichText component](https://github.com/WordPress/gutenberg/blob/50f2ae28bcb095ce7e2fe0b81e09149666159292/packages/block-editor/src/components/rich-text/native/test/index.native.js).

3. What automated tests I added (or what prevented me from doing so)

I plan to add tests in Gutenberg when enforcing a maximum size for displayed fonts.

<hr />

PR submission checklist:


UI Changes testing checklist:
